tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookAdapterTests.cs

```diff
@@ -0,0 +1,189 @@
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes("wrong-formatted-json"));
+
+            var httpRequest = new Mock<HttpRequest>();
+            httpRequest.SetupGet(req => req.Query[It.IsAny<string>()]).Returns("test");
+            httpRequest.SetupGet(req => req.Body).Returns(stream);
+            var httpResponse = new Mock<HttpResponse>();
+            var bot = new Mock<IBot>();
+            bot.Setup(x => x.OnTurnAsync(It.IsAny<TurnContext>(), It.IsAny<CancellationToken>())).Callback(() =>
+            {
+            });
+
+            await Assert.ThrowsAsync<Exception>(async () =>
+            {
+                await facebookAdapter.ProcessAsync(httpRequest.Object, httpResponse.Object, bot.Object, default(CancellationToken));
+            });
+        }
+
+        [Fact]
+        public async void ProcessAsyncShouldSucceed()
+        {
+            var payload = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\Payload.json");
+
+            var clientWrapper = new Mock<FacebookClientWrapper>(_testOptions);
+            clientWrapper.Setup(x => x.VerifySignature(It.IsAny<HttpRequest>(), It.IsAny<string>())).Returns(true);
+
+            var facebookAdapter = new FacebookAdapter(clientWrapper.Object);
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(payload));
+
+            var httpRequest = new Mock<HttpRequest>();
+            httpRequest.SetupGet(req => req.Query[It.IsAny<string>()]).Returns("test");
+            httpRequest.SetupGet(req => req.Body).Returns(stream);
+            var httpResponse = new Mock<HttpResponse>();
+            var bot = new Mock<IBot>();
+            bot.Setup(x => x.OnTurnAsync(It.IsAny<TurnContext>(), It.IsAny<CancellationToken>())).Callback(() =>
+            {
+            });
+
+            await facebookAdapter.ProcessAsync(httpRequest.Object, httpResponse.Object, bot.Object, default(CancellationToken));
+        }
+
+        [Fact]
+        public async void SendActivitiesAsyncShouldFailWithActivityTypeNotMessage()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var activity = new Activity
+            {
+                Type = ActivityTypes.Event,
+            };
+
+            Activity[] activities = { activity };
+
+            var turnContext = new TurnContext(facebookAdapter, activity);
+
+            await Assert.ThrowsAsync<Exception>(async () =>
+            {
+                await facebookAdapter.SendActivitiesAsync(turnContext, activities, default);
+            });
+        }
+
+        [Fact]
+        public async void SendActivitiesAsyncShouldSucceedWithActivityTypeMessage()
+        {
+            var clientWrapper = new Mock<FacebookClientWrapper>(_testOptions);
+            clientWrapper.Setup(x => x.GetApiAsync(It.IsAny<Activity>())).Returns(Task.FromResult(new FacebookClientWrapper(_testOptions)));
```

Comment:
there'd be a space between this variable and the following line. Also, change var X to a more expressive name